src/merkle/tiered_smt/values.rs

```diff
@@ -260,18 +271,23 @@ impl StoreEntry {
             }
         }
     }
+}
 
+impl<T> StoreEntry<T>
+where
+    T: Default,
+{
     /// Removes the key-value pair with the specified key from this entry, and returns the value
     /// of the removed pair. If the entry did not contain a key-value pair for the specified key,
     /// None is returned.
     ///
     /// If the last last key-value pair was removed from the entry, the second tuple value will
     /// be set to true.
-    pub fn remove(&mut self, key: &RpoDigest) -> (Option<Word>, bool) {
+    pub fn remove(&mut self, key: &RpoDigest) -> (Option<T>, bool) {
         match self {
             StoreEntry::Single(kv_pair) => {
                 if kv_pair.0 == *key {
-                    (Some(kv_pair.1), true)
+                    (Some(mem::take(&mut kv_pair.1)), true)
```

Comment:
This has slightly different semantics, but it should not be observable by any user.

Before the value was left instance, and the leaf was removed. Now, the value is set to ZERO, and the leaf is removed. I expect the compiler to remove the unnecessary write to the default.

There are a few alternatives:

- Add the representation and `Empty` variant to the structure, and use that instead, just to be more explicit
- Add an `Copy` bound on `T` to go back to the previous behavior
- Merge this back into the TSMT, so that we don't need the flags